cmd/spire-server/cli/run/run.go

```diff
@@ -654,22 +667,65 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 	return sc, nil
 }
 
-func parseBundleEndpointProfile(config federatesWithConfig) (trustDomainConfig *bundleClient.TrustDomainConfig, err error) {
-	// First check the number of bundle endpoint profiles in the config
-	objectList, ok := config.BundleEndpointProfile.(*ast.ObjectList)
-	if !ok {
-		return nil, errors.New("malformed configuration")
+func parseBundleEndpointConfigProfile(config *bundleEndpointConfig, dataDir string, log logrus.FieldLogger) (*bundle.ACMEConfig, error) {
+	switch {
+	case config.ACME != nil && config.Profile != nil:
+		return nil, errors.New("bundle endpoint 'acme' or 'profile' can be set")
+
+	case config.ACME != nil:
+		return configToACMEConfig(config.ACME, dataDir), nil
+
+	case config.Profile == nil:
+		log.Warn("bundle endpoint has no profile set, using https_spiffe as default, please choose a profile.")
+		return nil, nil
 	}
-	if len(objectList.Items) != 1 {
-		return nil, errors.New("exactly one bundle endpoint profile is expected")
+
+	// Profile is set, parse it
+	configString, err := parseSingleAstNode(config.Profile)
+	if err != nil {
+		return nil, err
 	}
 
-	// Parse the configuration
-	var data bytes.Buffer
-	if err := printer.DefaultConfig.Fprint(&data, config.BundleEndpointProfile); err != nil {
+	profileConfig := new(bundleEndpointConfigProfile)
+	if err := hcl.Decode(profileConfig, configString); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration: %w", err)
+	}
+
+	switch {
+	case profileConfig.HTTPSWeb != nil:
+		var acmeConfig *bundle.ACMEConfig
+
+		acme := profileConfig.HTTPSWeb.ACME
+		if acme != nil {
+			acmeConfig = configToACMEConfig(acme, dataDir)
+		}
+		return acmeConfig, nil
+
+	// For now ignore SPIFFE configuration
+	case profileConfig.HTTPSSPIFFE != nil:
+		return nil, nil
+
+	default:
+		return nil, errors.New(`no bundle endpoint profile defined; current supported profiles are "https_spiffe" and 'https_web"`)
```

Comment:
I think this would be a bit clearer if we talk about configuration explicitly. If we're hitting this, then a profile is set but it's unknown? It would be nice if we could also say what the value of the unknown profile was so the user can more easily find it. I don't think we should block on that though

```suggestion
		return nil, errors.New(`unknown bundle endpoint profile configured; current supported profiles are "https_spiffe" and 'https_web"`)
```